src/node/optimizer/entryAnalysisPlugin.ts

```diff
@@ -15,7 +19,9 @@ export function entryAnalysisPlugin(): Plugin {
             const facadeInfo = this.getModuleInfo(bundle.facadeModuleId)
             // this info is exposed by rollup commonjs plugin
             if (facadeInfo?.meta?.commonjs?.isCommonJS) {
-              analysis.isCommonjs[bundle.name] = true
+              const outputPath = path.join(cacheDir, bundle.fileName)
```

Comment:
Maybe `analysis.isCommonjs[bundle.facadeModuleId] = true` wil be better? It is file entry path for chunk. The `node_modules/.vite_opt_cache/react.js` is also not unique for same package with different versions.

Reply:
This will not allow to use alternative optimizers, this way you are coupling to rollup

The file path is relative to root, if it was another package it would be a different path, like ‘node_modules/intrinsic_dep/react.js’

Also notice that the current optimizer does not distinguish between packages with same name but different file locations but it will in the future

Reply:
> This will not allow to use alternative optimizers, this way you are coupling to rollup

The `bundle.facadeModuleId` is the file real absoulte path, it should can be used with other scenario.

Reply:
Ok, good idea, now this is how _analysis looks like

```json
{
  "isCommonjs": {
    "../node_modules/react/index.js": true,
    "../node_modules/react-dom/index.js": true,
    "../node_modules/prop-types/index.js": true
  }
}
```